packages/@aws-cdk/aws-apigatewayv2/lib/http/api.ts

```diff
@@ -168,6 +180,17 @@ export class HttpApi extends Resource implements IHttpApi {
         autoDeploy: true,
       });
     }
+
+    if (props?.defaultDomainMapping) {
+      const options = props.defaultDomainMapping;
+      new HttpApiMapping(this, `${options.domainName}${options.mappingKey}`, {
+        api: this,
+        domainName: options.domainName,
+        // use '$default' stage when we create a new API with defaultDomainMapping
+        stage: this.defaultStage!,
+      });
+
+    }
```

Comment:
This will fail if createDefaultStage is false but defaultDomainMapping is specified.

Move this entire block inside the previous `if` statement.